File: hw/palacios/u3/HW17Maze/js/view/MazePrinter.js
export class MazePrinter {
  print(maze) {
    const grid = maze.getGrid();
    const rows = maze.getRows();
    const cols = maze.getCols();

    let output = ' ';
    for (let j = 0; j < cols; j++) {
      output += ' _';
    }
    console.log(output);

    for (let i = 0; i < rows; i++) {
      let line = '|';
      for (let j = 0; j < cols; j++) {
        const room = grid[i][j];
        const bottom = room.bottomWall ? '_' : ' ';
        const right = room.rightWall ? '|' : ' ';

        if (i === 0 && j === 0) {
          line += 'S' + right; // Start
        } else if (i === rows - 1 && j === cols - 1) {
          line += 'E' + right; // End
        } else {
          line += bottom + right;
        }
      }
      console.log(line);
    }
  }
}

File: hw/palacios/u3/HW17Maze/Maze/src/main/java/ec/edu/espe/maze/view/MazePrinter.java
package ec.edu.espe.maze.view;

import ec.edu.espe.maze.model.Room;
import ec.edu.espe.maze.model.Maze;

/**
 *
 * @author Daniel Palacios
 */
public class MazePrinter {

    public void print(Maze maze) {
        Room[][] grid = maze.getGrid();
        int rows = maze.getRows();
        int cols = maze.getCols();

        System.out.print(" ");
        for (int j = 0; j < cols; j++) {
            System.out.print(" _");
        }
        System.out.println();

        for (int i = 0; i < rows; i++) {
            System.out.print("|");
            for (int j = 0; j < cols; j++) {
                Room room = grid[i][j];

                String bottom = room.bottomWall ? "_" : " ";

                String right = room.rightWall ? "|" : " ";

                if (i == 0 && j == 0) {
                    System.out.print("S" + right);
                } else if (i == rows - 1 && j == cols - 1) {
                    System.out.print("E" + right);
                } else {
                    System.out.print(bottom + right);
                }
            }
            System.out.println();
        }
    }
}
